src/react/components/pages/editorPage/editorPage.tsx

```diff
@@ -324,6 +328,30 @@ export default class EditorPage extends React.Component<IEditorPageProps, IEdito
         });
     }
 
+    private async updateProjectTagsFromAsset(asset: IAssetMetadata) {
+        const assetTags = new Set();
+        asset.regions.forEach((region) => region.tags.forEach((tag) => assetTags.add(tag)));
+
+        const newTags: ITag[] = this.props.project.tags ? [...this.props.project.tags] : [];
+        let updateTags = false;
+
+        assetTags.forEach((tag) => {
+            if (!this.props.project.tags || this.props.project.tags.length === 0 ||
+                !this.props.project.tags.find((projectTag) => tag === projectTag.name) ) {
+                newTags.push({
+                    name: tag,
+                    color: TagColors[newTags.length],
+                });
+                updateTags = true;
+            }
+        });
+
+        if (updateTags) {
+            const newProject = {...this.props.project, tags: newTags};
```

Comment:
why are we creating a new project? What will happen to the old one?

Reply:
If you see just next line you see we are calling the redux action to save the project.  Because of redux immutability it has to be a new project.  The redux/react flow will update back everything.